carma_wm/src/collision_detection.cpp

```diff
@@ -22,13 +22,15 @@ namespace carma_wm {
 
     namespace collision_detection {
 
-        std::vector<cav_msgs::RoadwayObstacle> WorldCollisionDetection(const cav_msgs::RoadwayObstacleList& rwol, const cav_msgs::TrajectoryPlan& tp, const geometry_msgs::Vector3& size, const geometry_msgs::Twist& veloctiy, const __uint64_t target_time) {
+        std::vector<cav_msgs::RoadwayObstacle> WorldCollisionDetection(const cav_msgs::RoadwayObstacleList& rwol, const cav_msgs::TrajectoryPlan& tp, 
+                                                                        const geometry_msgs::Vector3& size, const geometry_msgs::Twist& velocity, 
+                                                                        const __uint64_t target_time) {
 
-            ROS_DEBUG_STREAM("WorldCollisionDetection");
+            ROS_DEBUG_STREAM("WorldCollisionDetection at target_time: " << target_time); //TODO: if target_time won't be needed, remove it from param list
```

Comment:
Based on the function comments it seems like target_time is meant to let you sample only a portion [0, target_time] of the provided trajectory for collision which is a reasonable optimization to have, but this is a pretty inefficient function already as it has 
object_count * prediction_count * trajectory_point
square root calls. 

Adjusting target time would reduce only 1 of those 3 terms.

Probably could be removed and the function slated for rework.

Reply:
Removed & added TODO reminder to consider refactoring.